src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetAzurePrimaryDataStep.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.dataset.flight.delete;
+
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.azure.blobstore.AzureBlobStorePdao;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+public class DeleteDatasetAzurePrimaryDataStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteDatasetAzurePrimaryDataStep.class);
+
+  private final AzureBlobStorePdao azureBlobStorePdao;
+  private final FireStoreDao fileDao;
+  private final DatasetService datasetService;
+  private final UUID datasetId;
+  private final ConfigurationService configService;
+
+  public DeleteDatasetAzurePrimaryDataStep(
+      AzureBlobStorePdao azureBlobStorePdao,
+      FireStoreDao fileDao,
+      DatasetService datasetService,
+      UUID datasetId,
+      ConfigurationService configService) {
+    this.azureBlobStorePdao = azureBlobStorePdao;
+    this.fileDao = fileDao;
+    this.datasetService = datasetService;
+    this.datasetId = datasetId;
+    this.configService = configService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+    if (configService.testInsertFault(ConfigEnum.LOAD_SKIP_FILE_LOAD)) {
+      // If we didn't load files, don't try to delete them
+      fileDao.deleteFilesFromDataset(dataset, fireStoreFile -> {});
+    } else {
+      fileDao.deleteFilesFromDataset(dataset, azureBlobStorePdao::deleteFile);
+    }
+
+    // this fault is used by the DatasetConnectedTest > testOverlappingDeletes
+    if (configService.testInsertFault(ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT)) {
+      logger.info("DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT");
+      while (!configService.testInsertFault(
+          ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT)) {
+        logger.info("Sleeping for CONTINUE FAULT");
+        TimeUnit.SECONDS.sleep(5);
+      }
+      logger.info("DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT");
+    }
+
+    FlightMap map = context.getWorkingMap();
+    map.put(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.NO_CONTENT);
```

Comment:
It seems odd that this step is responsible for adding the status code. I guess that only one step in a flight can set this value? How do we decide which step owns doing this?

Reply:
That's just part of the particular flight's logic

Reply:
It looks like all flights must set `STATUS_CODE` if they want to return any status other than `HttpStatus.OK`. So it can only be set once in a flight or its value will be overwritten. In theory it could be used to return different values based on what the steps do, but the code doesn't make use of it.

So the fact that it's done here is arbitrary, other than the step assumes that it's the only step in this flight that's setting this value. It could be done at the end of the flight in a fixed step (`SetNoContentStatusStep`) and it would have the same effect.